### PR TITLE
fix aggregation of death message of SUT in case of instance death

### PR DESCRIPTION
**Pull request by dothebart**
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/11152/

### CODE REVIEW

File: js/client/modules/@arangodb/process-utils.js [hunk 1]
```diff
@@ -1082,7 +1082,12 @@ function checkArangoAlive (arangod, options) {
   const ret = res.status === 'RUNNING' && crashUtils.checkMonitorAlive(ARANGOD_BIN, arangod, options, res);
 
   if (!ret) {
-    print(Date() + ' ArangoD with PID ' + arangod.pid + ' gone:');
+    if (!arangod.hasOwnProperty('message')) {
+      arangod.message = '';
+    }
+    let msg = ' ArangoD of role [' + arangod.role + '] with PID ' + arangod.pid + ' is gone';
+    print(Date() + msg + ':');
+    arangod.message += (arangod.message.len === 0) ? '\n' : '' + msg + ' ';
```
Comment (jsteemann) [inline js/client/modules/@arangodb/process-utils.js:1090]: Isn't it `length` in JavaScript?
```suggestion
    arangod.message += (arangod.message.length === 0) ? '\n' : '' + msg + ' ';
```

Comment (KVS85) [inline js/client/modules/@arangodb/process-utils.js:1090]: That could leave a comment with the following structure:
```
LOG:
OTHER LOG
```

Will OTHER LOG be logically connected to LOG here?

File: js/client/modules/@arangodb/process-utils.js [hunk 2]
```diff
@@ -1096,9 +1101,13 @@ function checkArangoAlive (arangod, options) {
...
       arangod.exitStatus = res;
-      analyzeServerCrash(arangod, options, 'health Check  - ' + res.signal);
+      msg = 'health Check Signal(' + res.signal + ') ';
+      analyzeServerCrash(arangod, options, msg);
       serverCrashedLocal = true;
-      print(Date() + " checkArangoAlive: Marking crashy - " + JSON.stringify(arangod));
+      arangod.message += msg;
+      msg = " checkArangoAlive: Marking crashy";
+      arangod.message += msg;
+      print(Date() + msg + ' - ' + JSON.stringify(arangod));
```
Comment (KVS85) [inline js/client/modules/@arangodb/process-utils.js:1110]: Why `msg` is reused here?

Initially we will print `Date() + ' ArangoD of role [' + arangod.role + '] with PID ' + arangod.pid + ' is gone' + ':'`.

`arangod.message` finally will be
```
' ArangoD of role [' + arangod.role + '] with PID ' + arangod.pid + ' is gone ' + 'health Check Signal(' + res.signal + ') ' + " checkArangoAlive: Marking crashy"
```
or
```
'\n ' + 'health Check Signal(' + res.signal + ') ' + " checkArangoAlive: Marking crashy"
```

Afterwards we print `Date() + msg + ' - ' + JSON.stringify(arangod)`. `arangod` here will have `' ArangoD of role [' + arangod.role + '] with PID ' + arangod.pid + ' is gone ' + 'health Check Signal(' + res.signal + ') ' + " checkArangoAlive: Marking crashy"`.

What is the logic of interpretation?

File: js/client/modules/@arangodb/test-utils.js [hunk 3]
```diff
@@ -341,9 +341,13 @@ function performTests (options, testList, testname, runFn, serverOptions, startS
         } else {
           serverDead = true;
           continueTesting = false;
+          let msg = '';
+          if (results[te].hasOwnProperty('message') && results[te].message.len > 0) {
```
Comment (KVS85) [inline js/client/modules/@arangodb/test-utils.js:345]: Isn't it length in JavaScript?

File: js/client/modules/@arangodb/test-utils.js [hunk 4]
```diff
@@ -341,9 +341,13 @@ function performTests (options, testList, testname, runFn, serverOptions, startS
...
+            msg = results[te].message + ' - ';
+          }
           results[te] = {
             status: false,
-            message: 'server is dead: + ' + results[te].message
+            message: 'server is dead: ' + msg + instanceInfo.message
```
Comment (KVS85) [inline js/client/modules/@arangodb/test-utils.js:350]: How can `results[te].message` not have `results[te].message.length > 0`? Where is it set?

File: js/client/modules/@arangodb/process-utils.js [hunk 5]
```diff
@@ -2039,6 +2052,7 @@ function startInstance (protocol, options, addArgs, testname, tmpDir) {
   let rootDir = fs.join(tmpDir || fs.getTempPath(), testname);
 
   let instanceInfo = {
+    message: '',
```
[the remark this answers is not in the document]
Comment (dothebart) [reply js/client/modules/@arangodb/process-utils.js:2055]: its now there so we can aggregate failure messages on it.

File: js/client/modules/@arangodb/process-utils.js [hunk 6]
```diff
@@ -1131,7 +1140,11 @@ function checkInstanceAlive (instanceInfo, options) {
   }
   
   let rc = instanceInfo.arangods.reduce((previous, arangod) => {
-    return previous && checkArangoAlive(arangod, options);
+    let ret = checkArangoAlive(arangod, options);
+    if (!ret) {
+      instanceInfo.message += arangod.message;
```
[the remark this answers is not in the document]
Comment (dothebart) [reply js/client/modules/@arangodb/process-utils.js:1145]: Instance is the whole instance - data of a full cluster / leader follower. 
It contains an array of process instances, which are i.e. db-server.
If one (or more) fail the check, and now got error messages, its collected into the total instance.

File: js/client/modules/@arangodb/test-utils.js (hunk 4 above)
Comment (dothebart) [reply js/client/modules/@arangodb/test-utils.js:350]: If there has been a test-result with a failure, this may still be in there.

**Comment by dothebart** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/11175/

File: js/client/modules/@arangodb/process-utils.js (hunk 2 above)
Comment (dothebart) [reply js/client/modules/@arangodb/process-utils.js:1110]: simply msg is re-used. But its re-assigned a value, so its content doesn't add up.

**Comment by dothebart** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/11186/

**Comment by dothebart** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/11223/

**Pull request merged**